proposals/automating-releases.md

```diff
@@ -0,0 +1,185 @@
+---
+title: automating-releases
+authors:
+  - @estroz
+reviewers:
+  - @jmrodri
+  - @joelanford
+  - @theishah
+approvers:
+  - TBD
+creation-date: 2020-09-19
+last-updated: 2020-09-24
+status: implementable
+---
+
+# automating-releases
+
+## Release Signoff Checklist
+
+- \[ \] Enhancement is `implementable`
+- \[ \] Design details are appropriately documented from clear requirements
+- \[ \] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created [for the website][operator-sdk-doc]
+
+## Open Questions
+
+1. We currently build images for an architecture on machines of that architecture.
+This is not a requirement for Go binaries themselves, but might be for Ansible/Helm
+dependencies since they install binaries for particular architectures. What are
+these dependencies and can they be installed in a container during image build?
+1. This proposal encompasses all binaries/images currently built for an operator-sdk release.
+Once Ansible- and Helm-related code are split into their respective repos, will
+this release process follow?
+1. Continue using our [changelog generator][sdk-changelog-generator], or use goreleaser's
+[generator][goreleaser-changelog]?
+
+## Summary
+
+This proposal outlines an automated release pipeline for the operator-sdk repository
+that includes both binary and image builds. The [`goreleaser`][goreleaser] release tool
+will replace the existing combination of the release script `release.sh`, manual `git`
+steps, and manual release publishing with a one "click" solution. To users, no changes
+will occur since the release artifacts are not changing. To developers, the release
+process will become much simpler, hopefully to the point where a new contributor can
+run the release with little process knowledge.
+
+## Motivation
+
+Releasing the operator-sdk repo currently requires several steps that are mostly manual,
+with some automation around building binaries for all architectures supported. Tools like
+[`goreleaser`][goreleaser] can automate this process entirely by tagging code, and building
+and publishing binaries and images with one "click".
+
+## Goals
+
+- Remove all manual steps from the release process
+- Centralize release configuration
+- One-click releasing
+
+### Non-Goals
+
+- Change the content or constituency of output release artifacts in any way
+- Change the testing process in any way
+- Change our CI provider
+
```

Comment:
Part of the one-click release should trigger an update to the samples as well. Probably not needed for initial implementation, but it is probably worth stating explicitly

Reply:
I think we expect the samples to be up-to-date on every single commit and checked in CI similar to our CLI docs. Since the SDK version is no longer included in the samples' `go.mod` files. I don't think our release process will have an impact on the samples. I could totally be missing something though.

Reply:
Right, the samples should not change on a release since they were updated to their releasable state in the parent commit.